keras_retinanet/models/retinanet.py

```diff
@@ -330,42 +326,43 @@ def retinanet_bbox(
     These layers include applying the regression values to the anchors and performing NMS.
 
     Args
-        inputs      : keras.layers.Input (or list of) for the input to the model.
-        num_classes : Number of classes to classify.
-        name        : Name of the model.
-        *kwargs     : Additional kwargs to pass to the minimal retinanet model.
+        model             : RetinaNet model to append bbox layers to. If None, it will create a RetinaNet model using **kwargs.
+        anchor_parameters : Struct containing configuration for anchor generation (sizes, strides, ratios, scales).
+        name              : Name of the model.
+        *kwargs           : Additional kwargs to pass to the minimal retinanet model.
 
     Returns
-        A keras.models.Model which takes an image as input and outputs the result from each submodel on every pyramid level and a list of boxes.
+        A keras.models.Model which takes an image as input and outputs the detections on the image.
 
-        The order is as defined in submodels. Using default values the output is:
+        The order is defined as follows:
         ```
         [
-            regression, classification, boxes
+            boxes, scores, labels, other[0], other[1], ...
         ]
         ```
     """
-    model = retinanet(inputs=inputs, num_classes=num_classes, **kwargs)
+    if model is None:
+        model = retinanet(anchor_parameters=anchor_parameters, **kwargs)
```

Comment:
How about just passing only `anchor_parameters.num_anchors()` to the `retinanet` function? The less coupling we have the better.

Reply:
:+1: